### PR TITLE
Use Scrabble notation for blanks and invalid words

### LINKED ISSUE #255: Scrabble blanks could use Scrabble notation
At the moment, all scored Scrabble words are in uppercase ("ARGUMENT").

We could render blanks differently:

From [Wikipedia](https://en.wikipedia.org/wiki/Scrabble#Notation_system):
> When a blank tile is employed in the main word, the letter it has been chosen to represent is indicated with a lower case letter, or, in handwritten notation, with a square around the letter.

### LINKED ISSUE #256: Scrabble could cross-reference against the wordlist
The Scrabble module could check words against the wordlist.

You could use the [sowpods npm module](https://www.npmjs.com/package/sowpods) for this.

It could either reject an invalid word, or could follow standard Scrabble notation by outputting the invalid word followed by an asterisk (`SOWPODS*`)

### PR DESCRIPTION
**Pull request by georgewatson**
e.g. output for `xyzzy` is `XYZzY* scores 26 points!`

Closes #255 
Closes #256

### CODE REVIEW

**Comment by georgewatson** [comment]
I've moved to [pillowfication/pf-sowpods](https://github.com/pillowfication/pf-sowpods), which bundles a nicer search function, and also tried to access the list less.

I didn't notice any significant increase in memory usage when running the bot on my laptop with this module compared to without it, which implies we don't need to be too concerned about dragging the entire scrabble dictionary into RAM.

File: modules/scrabble.js [hunk 1]
```diff
@@ -1,33 +1,35 @@
+var sowpods = require('pf-sowpods')
```
Comment (zuzak) [inline modules/scrabble.js]:
```suggestion
const sowpods = require('pf-sowpods')
```

File: modules/scrabble.js [hunk 2]
```diff
@@ -44,10 +46,19 @@ function scoreLetter (letter, str) {
     }
   }
 }
+function scrabbleNotate (str) {
+  var word = str.toLowerCase()
```
Comment (zuzak) [inline modules/scrabble.js]:
```suggestion
  let word = str.toLowerCase()
```

File: modules/scrabble.js [hunk 3]
```diff
@@ -44,10 +46,19 @@ function scoreLetter (letter, str) {
...
+  for (var letter of [...new Set(word)]) {
+    for (var i = 0; i < bag[letter]['count']; i++) {
```
Comment (zuzak) [inline modules/scrabble.js]:
```suggestion
    for (let i = 0; i < bag[letter]['count']; i++) {
```

there's probably a better way of looping in the Array prototype somewhere but I'm not sure what

File: modules/scrabble.js (hunk 1 above)
Comment (zuzak) [reply modules/scrabble.js]: (although an Object as constant has limited value)

File: modules/scrabble.js [hunk 4]
```diff
@@ -44,10 +46,19 @@ function scoreLetter (letter, str) {
     }
   }
 }
+function scrabbleNotate (str) {
+  var word = str.toLowerCase()
+  for (var letter of [...new Set(word)]) {
```
Comment (georgewatson) [inline modules/scrabble.js]:
```suggestion
  for (const letter of [...new Set(word)]) {
```

File: modules/scrabble.js [hunk 5]
```diff
@@ -44,10 +46,19 @@ function scoreLetter (letter, str) {
...
+      word = word.replace(letter, letter.toUpperCase())
+    }
+  }
+  return word + (sowpods.verify(word) ? '' : '*')
+}
 function getPunctuation (bot, score) {
   const minScore = bot.config.get('scrabble.minScore')
   const exclThreshold = bot.config.get('scrabble.exclamationThreshold')
-  const numberOfMarks = Math.floor((score - minScore) / exclThreshold)
+  const numberOfMarks = Math.max(Math.floor((score - minScore) / exclThreshold), 0)
```
Comment (zuzak) [inline modules/scrabble.js]: What does `.max()` do here? If `numberOfMarks` is falsy the line below will handle it

Comment (georgewatson) [reply modules/scrabble.js]: The next line tries to `.repeat` that many times. If the value is negative, that doesn't work.

**Pull request merged**